[MLIR] [LLVM] ninja check-dynamatic failed

[0/1] Running the Dynamatic regression tests
FAIL: Dynamatic :: Dialect/Handshake/invalid.mlir (26 of 26)
******************** TEST 'Dynamatic :: Dialect/Handshake/invalid.mlir' FAILED ********************
Exit Code: 139

Command Output (stderr):
--
RUN: at line 1: ~/envs/dynamatic/build/bin/dynamatic-opt ~/envs/dynamatic/test/Dialect/Handshake/invalid.mlir --split-input-file --verify-diagnostics
+ ~/envs/dynamatic/build/bin/dynamatic-opt ~/envs/dynamatic/test/Dialect/Handshake/invalid.mlir --split-input-file --verify-diagnostics
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: ~/envs/dynamatic/build/bin/dynamatic-opt ~/envs/dynamatic/test/Dialect/Handshake/invalid.mlir --split-input-file --verify-diagnostics
1.      MLIR Parser: custom op parser 'handshake.func'
#0 0x000055b50b5cb277 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (~/envs/dynamatic/build/bin/dynamatic-opt+0xc6277)
#1 0x000055b50b5c90ce llvm::sys::RunSignalHandlers() (~/envs/dynamatic/build/bin/dynamatic-opt+0xc40ce)
#2 0x000055b50b5cbbfa SignalHandler(int) Signals.cpp:0:0
#3 0x00007fad1b4cf520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
~/envs/dynamatic/build/test/Dialect/Handshake/Output/invalid.mlir.script: line 1:  6162 Segmentation fault      (core dumped) ~/envs/dynamatic/build/bin/dynamatic-opt ~/envs/dynamatic/test/Dialect/Handshake/invalid.mlir --split-input-file --verify-diagnostics

--

********************
********************
Failed Tests (1):
  Dynamatic :: Dialect/Handshake/invalid.mlir


Testing Time: 0.13s
  Passed: 25
  Failed:  1
FAILED: test/CMakeFiles/check-dynamatic ~/envs/dynamatic/build/test/CMakeFiles/check-dynamatic 
cd ~/envs/dynamatic/build/test && /usr/bin/python3.10 ~/envs/dynamatic/polygeist/llvm-project/build/./bin/llvm-lit -sv ~/envs/dynamatic/build/test
ninja: build stopped: subcommand failed.

--- a/README.md
+++ b/README.md
@@ -80,5 +80,3 @@ The following instructions can be used to setup Dynamatic from source. If you in
     cd build
     ninja check-dynamatic
     ```
-
-    _**Important note:** If Dynamatic is built in Release mode, one of the tests will crash (see [#164](https://github.com/EPFL-LAP/dynamatic/issues/164)), hence it is recommended to run these tests in the Debug build._

--- a/test/lit.cfg.py
+++ b/test/lit.cfg.py
@@ -33,6 +33,9 @@ llvm_config.use_default_substitutions()
 
 # excludes: A list of directories to exclude from the testsuite.
 config.excludes = ['CMakeLists.txt', 'README.md']
+if config.cmake_build_type == "Release":
+    print("[WARNING] Skipping `invalid.mlir` in Release mode")
+    config.excludes.append("invalid.mlir")
 
 # Tweak the PATH to include the tools dir.
 llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)

--- a/test/lit.site.cfg.py.in
+++ b/test/lit.site.cfg.py.in
@@ -35,6 +35,7 @@ config.dynamatic_src_root = "@DYNAMATIC_SOURCE_DIR@"
 config.dynamatic_obj_root = "@DYNAMATIC_BINARY_DIR@"
 config.dynamatic_tools_dir = "@DYNAMATIC_TOOLS_DIR@"
 config.dynamatic_shlib_dir = "@LLVM_LIBRARY_OUTPUT_INTDIR@"
+config.cmake_build_type = "@CMAKE_BUILD_TYPE@"
 
 # Support substitution of the tools_dir with user parameters. This is
 # used when we can't determine the tool dir at configuration time.

